docs/Write-Plugin.md

```diff
@@ -0,0 +1,58 @@
+<h1 align="center">Fastify</h1>
+
+# How to write a good plugin
+First of all thank you for having decided to build a plugin for Fastify. Fastify is a *battery not included* framework and plugins are its strength, so thank you.<br>
+As you probably know, Fastify has been built since the beginning with performances and low overhead in mind, so if you are building a plugin you should keep a special eye on those. In this document we will analyze what characterizes a good quality plugin.
+
+*Need some inspiration? You can use the label ["plugin suggestion"](https://github.com/fastify/fastify/issues?q=is%3Aissue+is%3Aopen+label%3A%22plugin+suggestion%22) in our issue tracker!*
+
+## Code
+Fastify uses different techniques to optimize its code, many of them are documented in our Guides. We highly recommend you to read [the hitchhiker's guide to plugins](https://github.com/fastify/fastify/blob/master/docs/Plugins-Guide.md) to discover all the APIs you can use to build your plugin and learn how use them to the best possible.
+
+Have you got some question or are you seeking for a suggestion? We are more than happy to help you! Just open an issue in our [help](https://github.com/fastify/help) repository.
+
+Once you submit a plugin to our [ecosystem](https://github.com/fastify/fastify/blob/master/docs/Ecosystem.md) list, we will review your code and help you improve it if needed.
+
+## Documentation
+Documentation is extremely important, if your plugin is not well documented we will not accept it, and probably it is difficult that it will be used because of the lack of the documentation.<br>
+If you want to see some good examples on how document a plugin take a look at:
+- [`fastify-cookie`](https://github.com/fastify/fastify-cookie)
+- [`point-of-view`](https://github.com/fastify/point-of-view)
+- [`fastify-compress`](https://github.com/fastify/fastify-compress)
+- [`under-pressure`](https://github.com/fastify/under-pressure)
+- [`fastify-caching`](https://github.com/fastify/fastify-caching)
+
+## License
+You can license your plugin as you prefer, we do not enforce any kind of license.<br>
+Our recommendation is to license your code as [MIT](https://choosealicense.com/licenses/mit/), in this way no one will have legal concerns when using it.
```

Comment:
Let's not pretend to be lawyers by giving legal advice. Reword to:

> We prefer the [MIT license] because we think it allows more people to use the code freely. For a list of alternative licenses see the `[OSI list](https://opensource.org/licenses)` or GitHub's `[choosealicense.com](https://choosealicense.com/)`.